rest_framework_nested/relations.py

```diff
@@ -4,24 +4,31 @@
 These fields allow you to specify the style that should be used to represent
 model relationships with hyperlinks.
 """
+from __future__ import annotations
 from functools import reduce
+from typing import Any, TypeVar, Generic
 
-import rest_framework.relations
-from rest_framework.relations import ObjectDoesNotExist, ObjectValueError, ObjectTypeError
+from django.core.exceptions import ObjectDoesNotExist
+from django.db.models import Model
+from rest_framework.relations import HyperlinkedRelatedField, ObjectValueError, ObjectTypeError
 from rest_framework.exceptions import ValidationError
+from rest_framework.request import Request
 
 
-class NestedHyperlinkedRelatedField(rest_framework.relations.HyperlinkedRelatedField):
+T_Model = TypeVar('T_Model', bound=Model)
+
+
+class NestedHyperlinkedRelatedField(HyperlinkedRelatedField, Generic[T_Model]):
     lookup_field = 'pk'
     parent_lookup_kwargs = {
         'parent_pk': 'parent__pk'
     }
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
         self.parent_lookup_kwargs = kwargs.pop('parent_lookup_kwargs', self.parent_lookup_kwargs)
         super().__init__(*args, **kwargs)
 
-    def get_url(self, obj, view_name, request, format):
+    def get_url(self, obj: Model, view_name: str, request: Request, format: str | None) -> str | None:
```

Comment:
Idk if `NestedHyperlinkedRelatedField` is always about a `django.db.models.Model`.

Reply:
Yes, all `RelatedField`s are about models.

https://www.django-rest-framework.org/api-guide/relations/#serializer-relations

> Relational fields are used to represent model relationships. They can be applied to ForeignKey, ManyToManyField and OneToOneField relationships, as well as to reverse relationships, and custom relationships such as GenericForeignKey.

Reply:
btw I copied many of the signatures from `djangorestframework-stubs` where I'm one of the maintainers, it covers most of the API surface of DRF itself.

https://github.com/typeddjango/djangorestframework-stubs/blob/master/rest_framework-stubs/relations.pyi#L133